dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -817,7 +817,11 @@ private Response getFileByToolVersionID(String registryId, String versionIdParam
      */
     private static String computeURLFromEntryAndRequestURI(String entry, String selfPath) {
         String url = ToolsImplCommon.getUrlFromId(config, entry);
-        return url + selfPath.split(URLEncoder.encode(entry, StandardCharsets.UTF_8))[1];
+        String[] splitPath = selfPath.split(URLEncoder.encode(entry, StandardCharsets.UTF_8));
+        if (splitPath.length < 2) {
+            throw new CustomWebApplicationException("not found", HttpStatus.SC_NOT_FOUND);
```

Comment:
a  longer message or unique code may be useful for debugging for users who do double-encode